sklearn/metrics/_classification.py

```diff
@@ -134,6 +134,28 @@ def _check_targets(y_true, y_pred):
     return y_type, y_true, y_pred
 
 
+def _weighted_sum_1d(sample_score, sample_weight, normalize=False, xp=None):
```

Comment:
```suggestion
def _weighted_sum_1d(sample_score, sample_weight, normalize=False, xp=None):
    """Specialized _array_api._average for the 1D output case.

    Implements numpy-specific variants when the backing data is managed by
    numpy and delegate to the generic Array API _average function otherwise. 

    This function converts the result to a Python `float` to make the result
    array namespace and device agnostic.
    """
```

Reply:
@fcharras what do you think of the above suggestion?